src/lib/Preview.js

```diff
@@ -1882,6 +1884,28 @@ class Preview extends EventEmitter {
         const fileId = typeof fileIdOrFile === 'string' ? fileIdOrFile : fileIdOrFile.id;
         return getProp(this.previewOptions, `fileOptions.${fileId}.${optionName}`);
     }
+
+    /**
+     * Refresh the access token
+     *
+     * @private
+     * @return {Promise<string|Error>}
+     */
+    refreshToken = () => {
+        if (typeof this.previewOptions.token !== 'function') {
+            return Promise.reject(new Error('Token expired and cannot refresh token.'));
+        }
+        if (this.retryTokenCount >= RETRY_TOKEN_COUNT) {
+            return Promise.reject(new Error('Reach refreshing token limit.'));
+        }
+        this.retryTokenCount += 1;
+        return getTokens(this.file.id, this.previewOptions.token).then(tokenOrTokenMap => {
+            if (!tokenOrTokenMap || typeof tokenOrTokenMap === 'string') {
```

Comment:
It seems like `getTokens` resolves to a tokenMap, even if the token generator only returns a string, if so, then we can get rid of L1903-L1905